### PR TITLE
ENH address CR installment 2

### PR DESCRIPTION
**Pull request by beckermr**
This PR addresses CR installment 2.

### CODE REVIEW

**Comment by beckermr** [comment]
ok @rmjarvis this PR is ready for a good look.

File: src/ccl_background.c [hunk 1]
```diff
@@ -16,85 +16,42 @@
 INPUT: scale factor, cosmology
 TASK: Compute E(a)=H(a)/H0
 */
-static double h_over_h0(double a, ccl_cosmology * cosmo, int *status)
-{ 
-  // Check if massive neutrinos are present - if not, we don't need to compute their contribution
+static double h_over_h0(double a, ccl_cosmology * cosmo, int *status) {
```
[the remark this answers is not in the document]
Comment (beckermr) [reply src/ccl_background.c]: will fix.

**Comment by beckermr** [comment]
OK. This PR will be merged in a week if nobody has objections.

File: tests/ccl_test_cosmology.c [hunk 2]
```diff
@@ -41,36 +41,17 @@ CTEST_SETUP(cosmology) {
...
-  
-  // Pull ccl_parameters object out of ccl_cosmology
-  ccl_parameters params = (*cosmo).params;
-  
-  ASSERT_DBL_NEAR_TOL(params.Omega_c, data->Omega_c, 1e-10);
-  ASSERT_DBL_NEAR_TOL(params.w0, -1.0, 1e-10);
-  ASSERT_DBL_NEAR_TOL(params.wa, 0.0, 1e-10);
+  ccl_cosmology * cosmo = ccl_cosmology_create(params, config);
+
+  // IDK what to test here
```
Comment (philbull) [inline tests/ccl_test_cosmology.c]: This needs to be cleaned up.

Comment (beckermr) [reply tests/ccl_test_cosmology.c]: I can remove the comment, but otherwise there is not a lot to test. lol

Comment (beckermr) [reply tests/ccl_test_cosmology.c]: Do you have a specific suggestion @philbull ?

Comment (beckermr) [reply tests/ccl_test_cosmology.c]: I removed the comment. LMK if you have other requests!

**Comment by beckermr** [comment]
The test failure here is flake8 v 3.6.0 versus 3.5.0. Fuuuun. Fixing now.

**Comment by philbull** [comment]
Awesome, thanks Matt.

File: include/ccl_core.h [hunk 3]
```diff
@@ -193,96 +179,21 @@ ccl_cosmology * ccl_cosmology_create_with_lcdm_params(
...
- * @param status Status flag. 0 if there are no errors, nonzero otherwise.
- * For specific cases see documentation for ccl_error.c
- * @return void
- */
-ccl_parameters ccl_parameters_create_lcdm(double Omega_c, double Omega_b, double Omega_k, double h, double norm_pk, double n_s, int *status);
+/* ------- ROUTINE: ccl_parameters_create_flat_lcdm --------
+INPUT: some cosmological parameters needed to create a flat LCDM model
+TASK: call ccl_parameters_create to produce an LCDM model
+*/
+ccl_parameters ccl_parameters_create_flat_lcdm(double Omega_c, double Omega_b, double h,
```
Comment (elisachisari) [inline include/ccl_core.h]: So we've only kept this one out of all the specific cosmologies?

File: src/ccl_background.c [hunk 4]
```diff
@@ -17,127 +17,119 @@ INPUT: scale factor, cosmology
...
-ccl_species_nu_label <- massive neutrinos (physical)
+ccl_species_m_label <- matter
+ccl_species_l_label <- DE
+ccl_species_g_label <- radiation
+ccl_species_k_label <- curvature
+ccl_species_ur_label <- massless neutrinos
+ccl_species_nu_label <- massive neutrinos
 */
-double ccl_rho_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int is_comoving, int *status)
+double ccl_omega_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int *status)
```
Comment (elisachisari) [inline src/ccl_background.c]: This is an API change, it should be documented in the CHANGELOG. Checks pass, so I assume this was corrected in the python wrapping.

File: src/ccl_background.c [hunk 5]
```diff
@@ -17,127 +17,119 @@ INPUT: scale factor, cosmology
...
+ccl_species_crit_label <- critical (physical)
+ccl_species_m_label <- matter (physical)
+ccl_species_l_label <- DE (physical)
+ccl_species_g_label <- radiation (physical)
+ccl_species_k_label <- curvature (physical)
+ccl_species_ur_label <- massless neutrinos (physical)
+ccl_species_nu_label <- massive neutrinos (physical)
 */
-double ccl_omega_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int *status)
+double ccl_rho_x(ccl_cosmology * cosmo, double a, ccl_species_x_label label, int is_comoving, int *status)
```
Comment (elisachisari) [inline src/ccl_background.c]: More API changes.

File: src/ccl_background.c (hunk 4 above)
Comment (elisachisari) [reply src/ccl_background.c]: Oh, wait, did omega_x and rho_x simply switch order in the file?

File: src/ccl_background.c [hunk 6]
```diff
@@ -451,39 +448,41 @@ void ccl_cosmology_compute_distances(ccl_cosmology * cosmo, int *status)
...
+    // we are using the previous value as a guess here to help the root finder
+    // as long as we use small steps in a this should be fine
+    a_of_chi(y[i],cosmo, status, &a0,s);
     a[i]=a0;
   }
-  
+
   gsl_root_fdfsolver_free(s);
-  if(chistatus) {
+  if(splstatus) {
```
Comment (elisachisari) [inline src/ccl_background.c]: Is this really splstatus?

File: src/ccl_background.c [hunk 7]
```diff
@@ -833,98 +811,57 @@ double ccl_comoving_angular_distance(ccl_cosmology * cosmo, double a, int* statu
...
 void ccl_luminosity_distances(ccl_cosmology * cosmo, int na, double a[], double output[], int * status)
 {
-  if (!cosmo->computed_distances) {
-    ccl_cosmology_compute_distances(cosmo, status);
-    ccl_check_status(cosmo, status);
-  }
+  int _status;
+
   for (int i=0; i<na; i++) {
-    if ((a[i] > (1. - 1.e-8)) && (a[i] <= 1.))
```
Comment (elisachisari) [inline src/ccl_background.c]: So this check only happens in comoving_angular_distance now and all other distance functions call that one?

File: include/ccl_core.h (hunk 3 above)
Comment (beckermr) [reply include/ccl_core.h]: yes.

File: src/ccl_background.c (hunk 4 above)
Comment (beckermr) [reply src/ccl_background.c]: yes. :)

File: src/ccl_background.c (hunk 6 above)
Comment (beckermr) [reply src/ccl_background.c]: I changed the name of the variable and the code works. I don't follow the question though.

File: src/ccl_background.c (hunk 7 above)
Comment (beckermr) [reply src/ccl_background.c]: that is right.

File: src/ccl_background.c (hunk 6 above)
Comment (elisachisari) [reply src/ccl_background.c]: So you already check for splstatus in 418. Then in line 460 there is "status" from the a_of_chi(), but this doesn't get checked. Instead, you check for splstatus again. Was this the intention?

Comment (beckermr) [reply src/ccl_background.c]: Ack yes. It looks like this bug was in the code already. I changed the name of the parameter only. I will fix it.

**Comment by elisachisari** [comment]
Thanks for the update! Feel free to merge once the tests complete.

**Pull request merged**